packages/Kokkos/src/details/DTK_DetailsAlgorithms.hpp

```diff
@@ -22,6 +23,74 @@ void expand( Box &box, Box const &other );
 bool overlaps( Box const &box, Box const &other );
 // calculate the centroid of a box
 void centroid( Box const &box, Point &c );
+// Expands a 10-bit integer into 30 bits
+// by inserting 2 zeros after each bit.
+unsigned int expandBits( unsigned int v );
+
+// Calculates a 30-bit Morton code for the
+// given 3D point located within the unit cube [0,1].
+unsigned int morton3D( double x, double y, double z );
+
+#define __clz( x ) __builtin_clz( x )
+int clz( uint32_t x );
+int countLeadingZeros( unsigned int k );
+
+namespace Functor
+{
+using Box = AABB;
+
+class ExpandBoxWithBox
+{
+  public:
+    ExpandBoxWithBox( Box const *bounding_boxes )
+        : _greatest( Kokkos::ArithTraits<double>::max() )
+        , _lowest( -_greatest )
+        , _bounding_boxes( bounding_boxes )
+    {
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void init( Box &box )
+    {
+        for ( int d = 0; d < 3; ++d )
+        {
+            box[2 * d] = _greatest;
+            box[2 * d + 1] = _lowest;
+        }
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()( int const i, Box &box ) const
+    {
+        for ( int d = 0; d < 3; ++d )
+        {
+            box[2 * d] =
+                KokkosHelpers::min( _bounding_boxes[i][2 * d], box[2 * d] );
+            box[2 * d + 1] = KokkosHelpers::max( _bounding_boxes[i][2 * d + 1],
+                                                 box[2 * d + 1] );
+        }
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void join( volatile Box &dst, volatile Box const &src ) const
+    {
+        for ( int d = 0; d < 3; ++d )
+        {
+            // We need to access the underlying element directly because
+            // operator[] is not volatile
+            dst._minmax.m_elem[2 * d] = KokkosHelpers::min(
```

Comment:
That's problematic.  It was renamed to `m_internal_implementation_private_member_data ` in the Kokkos develop branch.